### DISCUSSION TITLE
Perbedaan dari plot berikut

### DISCUSSION THREAD

**Discussion by vlvivian** [teamalgoritma/community#71] [Q&A] [answered]
![Screenshot (214)](https://user-images.githubusercontent.com/103489378/164894753-8780c548-33c5-4f35-b7fd-2e8af0751558.png)
![Screenshot (215)](https://user-images.githubusercontent.com/103489378/164894755-4849a189-f223-4896-8bb9-d0524df306f6.png)
![Screenshot (216)](https://user-images.githubusercontent.com/103489378/164894756-15430c93-d33b-430c-a3a3-6838b34bcb3c.png)
![Screenshot (217)](https://user-images.githubusercontent.com/103489378/164894758-4137a848-ec6e-44a6-a276-b7c63e2b7777.png)
![Screenshot (213)](https://user-images.githubusercontent.com/103489378/164894760-136e501d-dd90-4f15-b0a4-2987dc99ab38.png)

Pak/Bu mau bertanya kenapa plot berikut bisa berbeda-beda, padahal terdapat coding yang hanya ditambahkan 'color' tetapi hasilnya menjadi berbeda ?

**Comment by onlyphantom** [comment, marked as answer]
It has nothing to do with `color` at all. Even if you don't specify `color`, `geom_jitter()` always has that random variation. The whole point was to take values like `c(4,4,6,7)` and instead of overlapping them in a straight line, **spread them out randomly** so they are not concentrated on a single line. 

If `c(4,4,6,7)` are plotted on a line, you will see the two `4` overlapping. Spreading the first `4` to the left and the second `4` to the right solves that issue.

To make the exact same `geom_jitter` plot every time, use a `seed`.

This is how you would do it:
```
ggplot(...) + 
    geom_boxplot(...)
    geom_point(position = position_jitter(seed = 123))
```

Now, every time you run the code, you get the exact same one because it is seeded. 

You can also `set.seed` outside the plotting:

```
set.seed(123)
ggplot(...) + 
    geom_boxplot(...)
    geom_point()
```

To understand more about what `seed` is doing for reproducibility, I explained this in a lot more details in this video (skip to 19:05):

https://www.youtube.com/watch?v=SyUBti7uMFk&t=1270s
